test/e2e/set/nodeconfig/nodeconfig_disksetup.go

```diff
@@ -149,30 +150,39 @@ var _ = g.Describe("Node Setup", framework.Serial, func() {
 		clientPod, err = controllerhelpers.WaitForPodState(waitCtx1, f.KubeClient().CoreV1().Pods(clientPod.Namespace), clientPod.Name, controllerhelpers.WaitForStateOptions{}, utils.PodIsRunning)
 		o.Expect(err).NotTo(o.HaveOccurred())
 
-		framework.By("Creating %d loop devices", numberOfDevices)
-		staticDevices, cleanup, err := makeLoopDevices(f.KubeClient().CoreV1(), clientPod, numberOfDevices)
-		o.Expect(err).NotTo(o.HaveOccurred())
-
-		defer func() {
-			err := cleanup()
-			o.Expect(err).NotTo(o.HaveOccurred())
-		}()
-
 		raidName := rand.String(8)
 		mountPath := fmt.Sprintf("/mnt/disk-setup-%s", f.Namespace())
 		hostMountPath := path.Join("/host", mountPath)
 
 		filesystem := scyllav1alpha1.XFSFilesystem
 		mountOptions := []string{"prjquota"}
 
+		loopDeviceNames := make([]string, 0, numberOfDevices)
+		for i := 0; i < numberOfDevices; i++ {
+			loopDeviceNames = append(loopDeviceNames, fmt.Sprintf("disk-%d", i))
+		}
+
 		nc.Spec.LocalDiskSetup = &scyllav1alpha1.LocalDiskSetup{
+			LoopDevices: func() []scyllav1alpha1.LoopDeviceConfiguration {
+				var ldcs []scyllav1alpha1.LoopDeviceConfiguration
+
+				for _, ldName := range loopDeviceNames {
+					ldcs = append(ldcs, scyllav1alpha1.LoopDeviceConfiguration{
+						Name:      ldName,
+						ImagePath: fmt.Sprintf("/mnt/%s.img", ldName),
+						Size:      resource.MustParse("32M"),
+					})
+				}
+
+				return ldcs
+			}(),
 			RAIDs: []scyllav1alpha1.RAIDConfiguration{
 				{
 					Name: raidName,
 					Type: scyllav1alpha1.RAID0Type,
 					RAID0: &scyllav1alpha1.RAID0Options{
 						Devices: scyllav1alpha1.DeviceDiscovery{
-							NameRegex:  strings.Join(staticDevices, "|"),
+							NameRegex:  `disk-\d+`,
```

Comment:
```suggestion
							NameRegex:  `^/dev/loops/disk-\d+$`,
```

Reply:
done

Reply:
Actually this wasn't supported because raid controller use `losetup` to list block devices which only have kernel names . Updated code to support both name used in API, and path to symlink so that both `disk-\d+` and `/dev/loops/disk-\d+` works as expected.

Reply:
> Actually this wasn't supported because raid controller use losetup to list block devices which only have kernel names 

I got this mixed with devices, sry

>  to support both name used in API

I'd rather avoid the ambiguity of matching one regex against two strings

Reply:
updated